src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/BookingTest.kt

```diff
@@ -3204,6 +3211,158 @@ class BookingTest : IntegrationTestBase() {
+          "http://api/events/cas3/person-departure-updated",
+        )
+      }
+    }
+  }
+
+  @Test
+  fun `Successfully send departed events when create departure is invoked for existing booking without departure detail`() {
+    `Given a User`(roles = listOf(UserRole.CAS3_ASSESSOR)) { userEntity, jwt ->
+      `Given an Offender` { offenderDetails, _ ->
+        val now = LocalDateTime.now()
+        val notes = "Some notes about the departure"
+        val premises = createTemporaryAccommodationPremises(userEntity)
+        val booking = createTemporaryAccommodationBooking(premises, offenderDetails)
+        val reasonEntity = departureReasonEntityFactory.produceAndPersist {
+          withServiceScope("temporary-accommodation")
+        }
+        val moveOnCategoryEntity = moveOnCategoryEntityFactory.produceAndPersist {
+          withServiceScope("temporary-accommodation")
+        }
+
+        callPersonDepartureDateAPIAndAssertDepartureDetail(
+          "/premises/${premises.id}/bookings/${booking.id}/departures",
+          buildNewDepartureDate(now, reasonEntity, moveOnCategoryEntity, notes),
+          booking,
+          now,
+          jwt,
+        )
+        assertPublishedSNSEvent(
+          booking,
+          "accommodation.cas3.person.departed",
+          "Someone has left a Transitional Accommodation premises",
+          "http://api/events/cas3/person-departed",
+        )
+      }
+    }
+  }
+
+  private fun assertPublishedSNSEvent(
+    booking: BookingEntity,
+    eventType: String,
+    eventDescription: String,
+    detailUrl: String,
+  ) {
+    val emittedMessage = inboundMessageListener.blockForMessage()
+
+    assertThat(emittedMessage.eventType).isEqualTo(eventType)
+    assertThat(emittedMessage.description).isEqualTo(eventDescription)
+    assertThat(emittedMessage.detailUrl).matches("$detailUrl/[0-9a-fA-F]{8}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{12}")
+    assertThat(emittedMessage.personReference.identifiers).containsExactlyInAnyOrder(
+      SnsEventPersonReference("CRN", booking.crn),
+      SnsEventPersonReference("NOMS", booking.nomsNumber!!),
+    )
+  }
+
+  private fun callPersonDepartureDateAPIAndAssertDepartureDetail(
+    url: String,
+    requestBody: NewDeparture,
+    booking: BookingEntity,
+    now: LocalDateTime,
+    jwt: String,
+  ) {
+    webTestClient.post()
+      .uri(url)
+      .header("Authorization", "Bearer $jwt")
+      .bodyValue(requestBody)
+      .exchange()
+      .expectStatus()
+      .isOk
+      .expectBody()
+      .jsonPath("$.id").isNotEmpty
+      .jsonPath("$.bookingId").isEqualTo(booking.id.toString())
+      .jsonPath("$.notes").isEqualTo("Some notes about the departure")
+      .jsonPath("$.dateTime").isEqualTo(now.atOffset(ZoneOffset.UTC).toString())
+      .jsonPath("$.reason").isNotEmpty
+      .jsonPath("$.moveOnCategory").isNotEmpty
+  }
+
+  private fun buildNewDepartureDate(
```

Comment:
This would probably be better named as `buildNewDeparture` as it's the `NewDeparture` class that's being built, not the date specifically.

Reply:
done